controllers/database_controller.go

```diff
@@ -191,9 +190,13 @@ func (r *DatabaseReconciler) deleteDatabase(dr *DatabaseReconciliation, instance
 	database := dr.resource
 
 	r.Log.V(1).Info("setup Stardog Client from ", "ref", instance)
-	auth, err := dr.reconciliationContext.initStardogClientFromRef(r.Client, instance)
-	if err != nil {
-		return err
+	auth, disabled, err := dr.reconciliationContext.initStardogClientFromRef(r.Client, instance)
+	if err != nil || disabled {
+		if err != nil {
+			return fmt.Errorf("cannot initialize stardog client: %v", err)
+		}
+		r.Log.Info("skipping resource from reconciliation", "instance", instance.Name, "resource", dr.resource.Name)
+		return nil
 	}
```

Comment:
You're doing two different things in the same if here... Checking for errors or if it's disabled. This could be confusing, I would maybe split this to different statements.